### DISCUSSION TITLE
Event Streams (Office Hours: Thu Dec 3)

### DISCUSSION THREAD

**Discussion by mindytom** [twilio-labs/paste#984] [Office hours] [answered]
**Background:** One stop shop for all types of Twilio events data (Messaging, Voice, Taskrouter, etc). Right now I'm designing the Console UI experience  
**Stage of Design:** Wires -> lo-fi 
**Link to prototype:** https://twilio.invisionapp.com/share/Y4ZF0UNEJRA 

- What's the recommended way to show statuses within a table? Specifically warnings and errors 
<img width="818" alt="Screen Shot 2020-12-01 at 11 05 02 AM" src="https://user-images.githubusercontent.com/65037539/100785126-15aa7a80-33c5-11eb-9e78-f176b3c6031a.png">

- What's the recommended pattern for editing? There doesn't seem to be a consistent pattern yet. Would love to learn what DSYS is thinking about this. 

- Feedback on component usage and interaction of overall Event Stream's first release.

**Comment by SiTaggart** [comment, marked as answer]
Hi @mindytom, as there are 3 pretty distinct areas it might be hard to get one answer. 

## Warnings and errors in tables

For showing warnings and errors in a table, one requirement from an accessibility point of view would be that the indicators are placed in the table itself, so that are associated with the offending row. Ideally this would be it's own dedicated column, but prepending them to the data in the row header (usually the first cell in each row) would be sufficient.

So in your example, the warning icon should either be in a column before the "sinks" column, or should be prepended to the SID/Identifier inside the table cell itself.

## Edit pattern 

Someone from the team will try to respond before your slot tomorrow regarding the pattern for editing. I suspect the answer will be that we don't have a position yet, but if you'd like to be the one to contribute it to the [Patterns section on the website](https://paste.twilio.design/patterns), we'd be grateful for it.

## General feedback

For component usage and interaction of overall experience, are there any specific areas you have concerns about that you'd like feedback on? The more specific the better feedback we can provide as general feedback on an entire experience is pretty broad and the team might not have enough time to dive that deeply.

**Reply by serifluous** [reply]
+1 to what Simon said about Editing. In console, it's generally the [Object details pattern](https://paste.twilio.design/patterns/object-details) + a form footer, but we've seen loads of variations. For now, I'd recommend hopping into a crit with more console designers to get their feedback.

And yes! We would definitely appreciate a pattern contribution.
